basic_pitch/inference.py

```diff
@@ -18,12 +18,37 @@
 import csv
 import enum
 import json
+import logging
 import os
 import pathlib
-from typing import Dict, List, Optional, Sequence, Tuple, Union
+from typing import Any, Dict, Iterable, List, Optional, Sequence, Tuple, Union, cast
+
+
+from basic_pitch import CT_PRESENT, ONNX_PRESENT, TF_PRESENT, TFLITE_PRESENT
+
+try:
+    import tensorflow as tf
+except ImportError:
+    pass
+
+try:
+    import coremltools as ct
+except ImportError:
+    pass
+
+try:
+    import tflite_runtime.interpreter as tflite
+except ImportError:
+    if TF_PRESENT:
+        import tensorflow.lite as tflite
+
+try:
+    import onnxruntime as ort
+except ImportError:
+    pass
 
-from tensorflow import Tensor, signal, keras, saved_model
 import numpy as np
+import numpy.typing as npt
```

Comment:
is it correct style to have a block of code running in between the imports?

Reply:
It isn't good style, but there's not really a better solution I could come up with. Generally I've seen this pattern done when you're dealing with extras that may or may not be installed.